src/topology/metric_space/dilation.lean

```diff
@@ -0,0 +1,390 @@
+/-
+Copyright (c) 2022 Hanting Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Dilations of emetric and metric spaces
+Authors: Hanting Zhang
+-/
+import topology.metric_space.antilipschitz
+import data.fun_like.basic
+
+/-!
+# Dilations
+
+We define dilations, i.e., maps between emetric spaces that satisfy
+`edist (f x) (f y) = r * edist x y`.
```

Comment:
This description is misleading because it doesn't allow `r = 0` or `r = \top`. I think we should add an explanation of *why* we don't want to consider those cases in this module docstring.